lib/vtls/vtls.c

```diff
@@ -1589,8 +1594,8 @@ CURLcode Curl_ssl_peer_init(struct ssl_peer *peer, struct Curl_cfilter *cf)
     }
 
     peer->sni = NULL;
-    peer->is_ip_address = is_ip_address(peer->hostname)? TRUE : FALSE;
-    if(peer->hostname[0] && !peer->is_ip_address) {
+    peer->type = get_peer_type(peer->hostname);
+    if(peer->type == CURL_SSL_PEER_DNS) {
```

Comment:
I notice you removed the 0 length hostname check for SNI, I assume because it's not possible

Reply:
Good catch. I re-added the check. I *believe* is is not necessary, but stranger things have happened before. Added a DEBUGASSERT to catch those situations in debug builds.